tests/diagnostics/attribute-error.slang.expected

```diff
@@ -0,0 +1,10 @@
+result code = 1
+standard error = {
+tests/diagnostics/attribute-error.slang(19): error 30019: expected an expression of type 'float', got 'String'
+tests/diagnostics/attribute-error.slang(11): note 31122: see defintion for attribute parameter 'MyStructAttribute.fParam'
```

Comment:
In this case the `see definition for attribute` message makes more sense. (Although note that there is a typo in the error message: defintion).

There is a general problem here that our type-coercion routine currently doesn't have any context passed down to explain where the expected type came from. Rather than patch that up here in an ad hoc fashion (which ends up also catching things like undefined identifiers), I'd rather strip out the note from this PR and have a separate PR add fundamental support to the "coerce" logic that allows some kind of context to be passed down that can emit the "see this thing" diagnostic as needed.

Reply:
Fixed. Removed this message.